src/initialize.F90

```diff
@@ -902,4 +916,230 @@ subroutine allocate_banks()
 
   end subroutine allocate_banks
 
+!===============================================================================
+! PREPARE_DISTRIBCELL initializes any distribcell filters present and sets the
+! offsets for distribcells
+!===============================================================================
+
+  subroutine prepare_distribcell()
+
+    integer :: i, j       ! Tally, filter loop counters
+    integer :: extra      ! Number of extra filters to add
+    integer :: n_filt     ! Number of filters originally in tally
+    logical :: count_all  ! Count all cells
+    type(TallyObject),    pointer :: tally            ! Current tally
+    type(Universe),       pointer :: univ             ! Pointer to universe
+    type(Cell),           pointer :: c                ! Pointer to cell
+    integer, allocatable :: univ_list(:)              ! Target offsets
+    integer, allocatable :: kounts(:,:)               ! Target count
+    logical, allocatable :: found(:,:)                ! Target found
+
+    count_all = .false.
+
+    ! Loop over tallies    
+    do i = 1, n_tallies
+
+      extra = 0
+
+      ! Get pointer to tally
+      tally => tallies(i)
+
+      n_filt = tally % n_filters
+
+      ! Loop over the filters to determine how many additional filters
+      ! need to be added to this tally
+      do j = 1, tally % n_filters
+
+        ! Determine type of filter
+        if (tally % filters(j) % type == FILTER_DISTRIBCELL) then
+          count_all = .true.
+          extra = extra + size(tally % filters(j) % int_bins) - 1
```

Comment:
Oh wow, very good point, I have no idea why the error check was written in this way. Just removed the ``extra`` variable and check for this situation in the way that you suggest.